tldw_Server_API/app/core/Agent_Client_Protocol/adapters/mcp_transports/stdio.py

```
@@ -0,0 +1,85 @@
+        self._connected = True
+
+    async def close(self) -> None:
+        """Shut down the subprocess and mark as disconnected."""
+        if self._client is not None:
+            await self._client.close()
+        self._connected = False
+
+    async def list_tools(self) -> list[dict[str, Any]]:
+        """Request the tool list from the MCP server."""
+        if self._client is None:
+            raise RuntimeError("Not connected")
```

Ruff (0.15.6)

[warning] Avoid specifying long messages outside the exception class

(TRY003)